Mathlib/Topology/ContinuousOn.lean

```diff
@@ -955,6 +955,13 @@ theorem ContinuousWithinAt.preimage_mem_nhdsWithin' {f : α → β} {x : α} {s
   h.tendsto_nhdsWithin (mapsTo_image _ _) ht
 #align continuous_within_at.preimage_mem_nhds_within' ContinuousWithinAt.preimage_mem_nhdsWithin'
 
+theorem ContinuousWithinAt.preimage_mem_nhdsWithin''
+    {f : α → β} {x : α} {y : β} {s t : Set β}
+    (h : ContinuousWithinAt f (f ⁻¹' s) x) (ht : t ∈ 𝓝[s] y) (hxy : y = f x) :
```

Comment:
Do you really want this extra `y` instead of just using `f x`?

Reply:
Yes. I started with the version without the `y`, and in all my applications I had first to rewrite `y` to put it exactly in the form of `f x` before applying the lemma. After 4 or 5 times, I decided it would be more efficient to change the lemma and to add a `y` there.

Reply:
Okay, I can convince myself that it makes sense since in this version of the lemma everything lives in the domain in some sense. So if it's useful let's keep it like that.